doc/modules/cassandra/pages/architecture/cql-on-accord.adoc

```diff
@@ -0,0 +1,612 @@
+allows reading at a requested consistency level. It largely inverts
+control of reading in Accord and uses Cassandra’s existing Read Executor
+functionality to determine what nodes to contact and what commands to
+send them while providing short read protection and blocking read
+repair. Read executors interface with Accord via the
+`https://github.com/apache/cassandra/blob/122f5300855d56131948575f80ce0594547c9040/src/java/org/apache/cassandra/service/reads/ReadCoordinator.java#L37[ReadCoordinator]`
+interface which can either send a regular read message or go through
+Accord to send an Accord specific read message which causes the read to
+execute at the appropriate command store in the appropriate
+transactional context after all dependencies have been applied.
+
+`ReadCoordinator` also intercepts blocking read repair during execution
+of an Accord transaction and executes it through the appropriate command
+store. The only legitimate way for this to occur is after Paxos key
+migration the data is only propagated at `QUORUM` so it is possible that
+Accord reading at `QUORUM` will find replicas to read repair. It’s not
+strictly necessary as we already know the data is propagated at
+`QUORUM`, but the support is there.
+
+`ReadCoordinator` also helps apply read repair mutations via Accord in
+`TransactionalMode.MIXED` and during migration by applying the read
+repair mutations in Accord’s execute phase instead of waiting for apply.
+This is safe because read repair only proposes already committed Accord
+writes or already unsafe non-SERIAL writes which aren’t allowed anyways.
+
+`https://github.com/apache/cassandra/blob/122f5300855d56131948575f80ce0594547c9040/src/java/org/apache/cassandra/service/accord/interop/AccordInteropPersist.java#L48[AccordInteropPersist]`
+adds support for synchronous commit and commit at a requested
+consistency level. It sends `AccordInteropApply` which is a synchronous
+apply message that only responds once application is complete.
+
+https://github.com/apache/cassandra/blob/122f5300855d56131948575f80ce0594547c9040/src/java/org/apache/cassandra/service/consensus/TransactionalMode.java#L34[`TransactionalMode`]
+defines the supported modes and
+`https://github.com/apache/cassandra/blob/122f5300855d56131948575f80ce0594547c9040/src/java/org/apache/cassandra/service/consensus/TransactionalMode.java#L34[commitCLForMode]`
+determines the commit consistency level and
+`https://github.com/apache/cassandra/blob/122f5300855d56131948575f80ce0594547c9040/src/java/org/apache/cassandra/service/consensus/TransactionalMode.java#L170[readCLForMode]`
+determines the read consistency level. These two methods take into
+account both the requested consistency level, the table specific
+migration state, the current transactional mode, and the target
+transactional mode in order to decide whether to honor the requested
+consistency level.
+
+=== Routing requests during migration
+
+During migration requests race with changes to
```

Comment:
Would benefit with a comma `During migration, requests` to disambiguate it's just generic requests, not a specific migration request.